src/Common/StringUtilities.h

```diff
@@ -0,0 +1,41 @@
+/*****************************************************************************
+ * Copyright (c) 2018-2022 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include <algorithm>
+#include <string>
+
+namespace openblack
+{
+inline bool EndsWith(std::string const& str, std::string const& ending)
+{
+	if (ending.size() > str.size())
+	{
+		return false;
+	}
+
+	return std::equal(ending.rbegin(), ending.rend(), str.rbegin());
+}
+
+inline bool BeginsWith(std::string const& str, std::string const& ending)
+{
+	if (ending.size() > str.size())
+	{
+		return false;
+	}
+
+	return std::equal(ending.begin(), ending.end(), str.begin());
+}
+
+inline void LowerCase(std::string& str)
```

Comment:
We already do this in here.
https://github.com/openblack/openblack/blob/33e0a1213173001167f2c4ace5afc07a40ea2dbb/src/Game.cpp#L466-L467

Could you use that formulation which is more C++ than C. And refactor it here.

Reply:
Also, the name sounds more like it should return a LowerCase version, not modify it. `ToLowerCase` would be better for inplace.

To be complete, it would be good to do the same for Upper.
Also a function for first letter capitalization which we do here: https://github.com/openblack/openblack/blob/e8e8431c0f813fdec2c089d609dab241f7c8ad0d/src/3D/Sky.cpp#L46